akka-actor-typed/src/main/scala/akka/actor/typed/internal/DelayedStart.scala

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (C) 2009-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.actor.typed.internal
+
+import akka.actor.typed.scaladsl.MutableStashBuffer
+import akka.actor.typed.{Behavior, scaladsl}
+import akka.annotation.InternalApi
+
+/**
+ * INTERNAL API
+ *
+ * A behavior that wraps another behavior, stashing all incoming messages and signals until
+ * it is sent the `Start` message, it then unstashes all messages and signals and switches to the delayedBehavior
+ */
+@InternalApi
+private[akka] object DelayedStart {
+
+  case object Start
+
+  // note that we cannot use StashBuffer here as we need to buffer signals as well as messages
+  def delayStart[T](delayedBehavior: Behavior[T]): Behavior[Any] =
+    scaladsl.Behaviors.deferred[Any] { _ ⇒
+      val buffer = MutableStashBuffer[Any](100)
```

Comment:
we could be somewhat more generous with the capacity, to avoid false failures, 10000?

Reply:
Yep, sure better safe than sorry. I think this should pretty much never get more than a single PostStop, Terminated etc. though as there can be no user space children started until this deferred behavior is started.